azurelinuxagent/ga/monitor.py

```diff
@@ -207,7 +208,7 @@ def init_sysinfo(self):
             self.sysinfo.append(TelemetryEventParam('VMId',
                                                     vminfo.vmId))
             self.sysinfo.append(TelemetryEventParam('ImageOrigin',
-                                                    vminfo.image_origin))
+                                                    int(vminfo.image_origin)))
```

Comment:
do the conversion at the point where image_origin is initialized?

Reply:
The values are already INT. The casting at the event level is making sure that the event does not evert get some ambiguous values in the future.

```
IMDS_IMAGE_ORIGIN_UNKNOWN = 0
IMDS_IMAGE_ORIGIN_CUSTOM = 1
IMDS_IMAGE_ORIGIN_ENDORSED = 2
IMDS_IMAGE_ORIGIN_PLATFORM = 3
```